includes/compat/trait-wc-stripe-subscriptions.php

```diff
@@ -62,6 +65,80 @@ public function maybe_init_subscriptions() {
 		add_action( 'template_redirect', [ $this, 'restore_order_pay_var' ], 101 );
 	}
 
+	/**
+	 * Add the necessary information to payment intents for subscriptions to allow Stripe to create
+	 * mandates for 3DS payments in India. It's ok to apply this across the board; Stripe will
+	 * take care of handling any authorizations.
+	 *
+	 * @param Array    $request          The HTTP request that will be sent to Stripe to create the payment intent.
+	 * @param WC_Order $order            The renewal order.
+	 * @param Array    $prepared_source  The source object.
+	 */
+	public function add_subscription_information_to_intent( $request, $order, $prepared_source ) {
+		// Just in case the order doesn't contain a subscription we return the base request.
+		if ( ! $this->has_subscription( $order->get_id() ) ) {
+			return $request;
+		}
+
+		// TODO: maybe this isn't necessary since this function should really only be called
+		//       when creating the intent? It's called in process_subscription_payment though
+		//       so it's probably needed here too?
+		// If we've already created a mandate for this order; use that.
+		$mandate = $order->get_meta( '_stripe_mandate_id', true );
+		if ( isset( $mandate ) && '' !== $mandate ) {
+			$request['mandate'] = $mandate;
+			return $request;
+		}
+
+		// TODO: what happens if there's more than one renewal order?
+		$renewals = wcs_get_subscriptions_for_renewal_order( $order->get_id() );
+		if ( 1 === count( $renewals ) ) {
+			$renewal_order   = reset( $renewals );
+			$parent_order_id = $renewal_order->get_parent_id();
+			$parent_order    = wc_get_order( $parent_order_id );
+
+			$mandate = $parent_order->get_meta( '_stripe_mandate_id', true );
+			if ( isset( $mandate ) && '' !== $mandate ) {
+				$request['mandate'] = $mandate;
+				return $request;
+			}
+		}
+
+		// Otherwise add the parameters required to create a mandate.
+
+		$subscriptions = wcs_get_subscriptions_for_order( $order->get_id() );
+
+		// If there are no subscriptions we just return.
+		if ( 0 === count( $subscriptions ) ) {
+			return $request;
+		}
+
+		// Get the first subscription associated with this order.
+		$sub = reset( $subscriptions );
+
+		if ( 1 === count( $subscriptions ) ) {
+			$request['payment_method_options']['card']['mandate_options']['amount_type']    = 'fixed';
+			$request['payment_method_options']['card']['mandate_options']['interval']       = $sub->get_billing_period();
+			$request['payment_method_options']['card']['mandate_options']['interval_count'] = $sub->get_billing_interval();
+		} else {
+			// If there are multiple subscriptions the amount_type becomes 'maximum' so we can charge anything
+			// less than the order total, and the interval is sporadic so we don't have to follow a set interval.
+			$request['payment_method_options']['card']['mandate_options']['amount_type'] = 'maximum';
+			$request['payment_method_options']['card']['mandate_options']['interval']    = 'sporadic';
+		}
```

Comment:
I'm not sure if this is the right approach since the orders actually don't fail in test mode when the mandate options are configured incorrectly. As such, there is no way to verify whether this is the correct approach or not.

We'll have to wait for further direction from Stripe so we can test this.